encoders/firebase-decoders-json/src/json/java/com/google/firebase/decoders/json/CreationContextImpl.java

```diff
@@ -0,0 +1,101 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.decoders.json;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import com.google.firebase.decoders.CreationContext;
+import com.google.firebase.decoders.FieldRef;
+import java.util.HashMap;
+
+public class CreationContextImpl implements CreationContext {
+  private HashMap<FieldRef<?>, Object> ctx = new HashMap<>();
+
+  CreationContextImpl() {}
+
+  void put(@NonNull FieldRef<?> ref, @Nullable Object val) {
+    ctx.put(ref, val);
+  }
+
+  // TODO: avoid auto-boxing for all primitive types
+  void put(@NonNull FieldRef<?> ref, int val) {
+    ctx.put(ref, val);
+  }
+
+  void put(@NonNull FieldRef<?> ref, short val) {
+    ctx.put(ref, val);
+  }
+
+  void put(@NonNull FieldRef<?> ref, long val) {
+    ctx.put(ref, val);
+  }
+
+  void put(@NonNull FieldRef<?> ref, float val) {
+    ctx.put(ref, val);
+  }
+
+  void put(@NonNull FieldRef<?> ref, double val) {
+    ctx.put(ref, val);
+  }
+
+  void put(@NonNull FieldRef<?> ref, boolean val) {
+    ctx.put(ref, val);
+  }
```

Comment:
Since these auto-box anyway, consider removing these methods altogether for now.